prow/cmd/marque/main.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os/exec"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/kube"
+)
+
+const (
+	// Time before generating the first cert, for safety.
+	sleepTime = time.Minute
+	// Time between renewals.
+	renewTime = 12 * time.Hour
+)
+
+func main() {
+	kc, err := kube.NewClientInCluster("default")
+	if err != nil {
+		logrus.WithError(err).Fatal("Error getting kube client.")
+	}
+
+	root, err := ioutil.TempDir("", "certbot")
+	if err != nil {
+		logrus.WithError(err).Fatal("Could not create temp dir.")
+	}
+	http.Handle("/", http.FileServer(http.Dir(root)))
+	go http.ListenAndServe(":http", nil)
+
+	logrus.Infof("Sleeping for %v before generating cert.", sleepTime)
+	time.Sleep(sleepTime)
+	if err := generate(root); err != nil {
```

Comment:
IIUC whenever the pod gets recreated, we'll generate a new cert?

seems a little wasteful, but I guess certs are free.